### PR TITLE
(Feature) lutaml_uml_datamodel_description macro

### PR DESCRIPTION
**Pull request by w00lf**
#34

### CODE REVIEW

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+.UML diagram of the {{ package.name }} Model.
+
+image::figures/{{ package.name }}.png[align="center"]
+
+{% elsif diagram.name contains ade_types_image_name %}
+The ADE data types provided for the {{ package.name }} module are
+illustrated in <<{{ package_name }}-uml-ade-types>>.
+
+[[{{ package_name }}-uml-ade-types]]
+.ADE classes of the CityGML {{ package.name }} module.
```
Comment (ronaldtse) [inline lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:42]: "CityGML" is not generic, it's the name of a package.

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+{% capture requirement_boundaries_filename %}requirements/{{ package_name }}/REQ_{{ package_name }}_Boundaries{% endcapture %}
+{% capture requirement_boundaries_content %}{% include requirement_boundaries_filename %}{% endcapture %}
+{% capture requirement_ade_use_filename %}requirements/{{ package_name }}/REQ_{{ package_name }}_ADE_Use{% endcapture %}
+{% capture requirement_ade_use_content %}{% include requirement_ade_use_filename %}{% endcapture %}
+{% unless requirement_classes_filename contains "Liquid error" and requirement_boundaries_content contains "Liquid error" and requirement_ade_use_content contains "Liquid error" %}
+{{equalsigns}}= Requirements
+
+{% unless requirement_classes_content contains "Liquid error" %}
+The following requirement defines the rules governing implementation
+of the CityGML Bridge Module as an Implementation Specification.
```
Comment (ronaldtse) [inline lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:74]: This is not generic text.

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+|{{ klass.definition }}
+
+{% endfor %}
+
+|===
+{% endif %}
+
+{{equalsigns}}= Additional Information
+
+Additional information about the {{ package.name }} can be found in the link:http://docs.opengeospatial.org/DRAFTS/20-066.html#ug-model-{{ package_name }}-section[OGC CityGML 3.0 Users Guide]
```
Comment (ronaldtse) [inline lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:140]: As above.

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages_data_dictionary.liquid [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+!{nbsp}{nbsp}{nbsp}{nbsp}Definition: ! {{ data_type.definition }}
+!{nbsp}{nbsp}{nbsp}{nbsp}Subclass of: ! {{ generalizations }}
+!{nbsp}{nbsp}{nbsp}{nbsp}Stereotype: !  «DataType»
+!===
+{% if data_type.associations.size > 0 %}
+{% capture rendered_associations %}
+{% for assoc in data_type.associations %}
+{% if assoc.member_end_attribute_name.size > 0 %}
+{% capture cardinality %}{% if assoc.member_end_cardinality.min == 'C' %}[0..{{ assoc.member_end_cardinality.max }}]{% elsif assoc.member_end_cardinality.min == 'M' and assoc.member_end_cardinality.max == '1' %}[1..1]{% else %}[{{ assoc.member_end_cardinality.max }}]{% endif %}{% endcapture %}
+!{set:cellbgcolor:#FFFFFF} {{ assoc.member_end_attribute_name }}  !<<{{assoc.member_end}}-section,{{assoc.member_end}}>> {{ cardinality }} ! {{ assoc.definition }}
```
Comment (ronaldtse) [inline lib/metanorma/plugin/lutaml/liquid_templates/_packages_data_dictionary.liquid:65]: We need to remove all non-Metanorma styles like `!{set:cellbgcolor:#FFFFFF}`

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid (hunk 1 above)
Comment (w00lf) [reply lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:42]: Removed diagram blocks with it

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid (hunk 2 above)
Comment (w00lf) [reply lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:74]: Removed

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid (hunk 3 above)
Comment (w00lf) [reply lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:140]: Removed

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages_data_dictionary.liquid (hunk 4 above)
Comment (w00lf) [reply lib/metanorma/plugin/lutaml/liquid_templates/_packages_data_dictionary.liquid:65]: Done, removed

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+{{ requirement_content }}
+{% endunless %}
+
+{% for diagram in package.diagrams %}
+{% capture codelists_image_name %}{{ package.name }} - Code lists{% endcapture %}
+{% capture ade_types_image_name %}{{ package.name }} - ADE Data types{% endcapture %}
+{% if diagram.name == package.name %}
+The UML diagram of the {{ package.name }} module is depicted in
+<<{{ package_name }}-uml>>.
+A detailed discussion of Requirements Class {{ package.name }}
```
Comment (ronaldtse) [inline lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:28]: This line needs to be removed.

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+can be found in the
+link:http://docs.opengeospatial.org/DRAFTS/20-066.html#ug-model-{{ package_name }}-section[CityGML 3.0 Users Guide].
+
+[[{{ package_name }}-uml]]
+.UML diagram of the {{ package.name }} Model.
+
+image::figures/{{ package.name }}.png[align="center"]
+
+{% elsif diagram.name contains ade_types_image_name %}
+The ADE data types provided for the {{ package.name }} module are
```
Comment (ronaldtse) [inline lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:38]: Remove.

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+[[{{ package_name }}-uml-codelists]]
+.Codelists from the CityGML {{ package.name }} module.
+image::figures/{{ package.name }}-Codelists.png[]
+{% endif %}
+{% endfor %}
+
+{% capture boundaries_filename %}boundaries/boundaries_{{ package_name }}{% endcapture %}
+{% capture boundaries_content %}{% include boundaries_filename %}{% endcapture %}
+{% unless boundaries_content contains "Liquid error" %}
+<<{{ package_name }}-boundaries-table>> lists the surfaces that are allowed as thematic surface boundaries of the space classes defined in the {{ package.name }} module:
```
Comment (ronaldtse) [inline lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:58]: This line is not generic.

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+{{ requirement_classes_content }}
+{% endunless %}
+{% unless requirement_boundaries_content contains "Liquid error" %}
+The implementation of this Module is further constrained by the
+following spatial boundary requirement:
+
+{{ requirement_boundaries_content }}
+{% endunless %}
+{% unless requirement_ade_use_content contains "Liquid error" %}
+The use of extension capabilities by Bridge elements is constrained
```
Comment (ronaldtse) [inline lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:85]: "Bridge" is CityGML specific.

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages_classes.liquid [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+!{nbsp}{nbsp}{nbsp}{nbsp}Definition: ! {{ klass.definition }}
+!{nbsp}{nbsp}{nbsp}{nbsp}Subclass of: ! {{ generalizations }}
+!{nbsp}{nbsp}{nbsp}{nbsp}Stereotype: !  «{{ klass.stereotype }}»
+!===
+{% if klass.associations.size > 0 %}
+{% capture rendered_associations %}
+{% for assoc in klass.associations %}
+{% if assoc.member_end_attribute_name.size > 0 %}
+{% capture cardinality %}{% if assoc.member_end_cardinality.min == 'C' %}[0..{{ assoc.member_end_cardinality.max }}]{% elsif assoc.member_end_cardinality.min == 'M' and assoc.member_end_cardinality.max == '1' %}[1..1]{% else %}[{{ assoc.member_end_cardinality.max }}]{% endif %}{% endcapture %}
+!{set:cellbgcolor:#FFFFFF} {{ assoc.member_end_attribute_name }}  !<<{{assoc.member_end}}-section,{{assoc.member_end}}>> {{ cardinality }} ! {{ assoc.definition }}
```
Comment (ronaldtse) [inline lib/metanorma/plugin/lutaml/liquid_templates/_packages_classes.liquid:18]: Remove all colors.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid (hunk 7 above)
Comment (w00lf) [reply lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:26]: Removed

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid (hunk 8 above)
Comment (w00lf) [reply lib/metanorma/plugin/lutaml/liquid_templates/_packages.liquid:52]: Commented out

File: lib/metanorma/plugin/lutaml/liquid_templates/_packages_classes.liquid (hunk 9 above)
Comment (w00lf) [reply lib/metanorma/plugin/lutaml/liquid_templates/_packages_classes.liquid]: Done

**Pull request merged**